core/execution/engine.go

```diff
@@ -215,8 +215,13 @@ func (e *Engine) Hash() []byte {
 	}
```

Comment:
If you can be bothered because its unrelated: this loop above:
```
for _, m := range e.markets {
	hash := m.Hash()
	e.log.Debug("market app state hash", logging.Hash(hash), logging.String("market-id", m.GetID()))
	hashes = append(hashes, string(hash))
}
```
if over a map, so between nodes the logs of the market hashes are sometimes in different orders which always caused me endless pain when comparing logs during snapshot bugs.

Wondering if it makes sense to use `e.marketsCpy` instead.

Reply:
sounds good. will do.

Reply:
addressed